src/AutoDark.jl

```diff
@@ -2,20 +2,19 @@ module AutoDark
 
 DARK_NOTIFY_RUNNING = false
 
-function auto_dark_apple(fun::Function)
+function proc_line_based(fun::Function, cmd::Cmd, output::AbstractDict)
     AutoDark.DARK_NOTIFY_RUNNING && return
 
-    cmd = `dark-notify`
     pin = Base.PipeEndpoint()
     pout = Base.PipeEndpoint()
     perr = Base.PipeEndpoint()
     proc = run(cmd, pin, pout, perr, wait=false)
-    process_running(proc) || error("Could not start dark-notify")
+    process_running(proc) || error("Could not start $cmd")
     AutoDark.DARK_NOTIFY_RUNNING = true
-    @async try
```

Comment:
I used `Threads.@spawn` instead of `@async` here because of this warning in the docs:
https://docs.julialang.org/en/v1/base/parallel/#Base.@async